src/compiler/crystal/semantic/call.cr

```diff
@@ -1058,7 +1058,7 @@ class Crystal::Call
       args["self"] = MetaVar.new("self", self_type)
     end
 
-    strict_check = body.is_a?(Primitive) && body.name == "proc_call"
+    strict_check = body.is_a?(Primitive) && (body.name == "proc_call" || body.name == "pointer_set")
```

Comment:
Wouldn't it make sense to remove this variable as its only used in the conditional which could be replaced by `case body.as?(Primitive).try &.name`?

Reply:
I would like to make it clear a strict check is happening.

Reply:
Or maybe yes, I don't know. I don't think it matters much, plus I won't be at the computer for a while